### PR TITLE
Support PostgreSQL datastore

### LINKED ISSUE #1135: Support event datastore
Sensu Go supports PostgreSQL for event datastore: https://docs.sensu.io/sensu-go/5.11/reference/datastore/

To help sites get started it would be handy to also manage PostgreSQL with Puppetlab's module so sites.

### PR DESCRIPTION
**Pull request by treydock**
# Pull Request Checklist

## Description

Add support for PostgreSQL datastore.

## Related Issue

Fixes #1135 

## Motivation and Context

Sensu Go 5.10 added support for PostgreSQL datastore that has to be setup with `sensuctl`. This adds support for that as well as optionally manages the actual postgresql database.

### CODE REVIEW

File: .fixtures.yml [hunk 1]
```diff
@@ -10,6 +10,12 @@ fixtures:
       repo: git://github.com/puppetlabs/puppetlabs-yumrepo_core
       ref: 1.0.1
       puppet_version: ">= 6.0.0"
+    # Need by postgresql
+    concat:
+      repo: git://github.com/puppetlabs/puppetlabs-concat.git
```
Comment (ghoneycutt) [inline .fixtures.yml]: this should use a tag

File: spec/acceptance/06_postgresql_spec.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
+require 'spec_helper_acceptance'
+
+describe 'sensu::backend class', unless: RSpec.configuration.sensu_cluster do
```
Comment (ghoneycutt) [inline spec/acceptance/06_postgresql_spec.rb]: How about a test that shows that after a client runs, there is expected data in the datastore. This would allow us to test from the user perspective and verify that the connection between the sensu backend and postgres are working as intended.

File: .fixtures.yml (hunk 1 above)
Comment (treydock) [reply .fixtures.yml]: Added a tag here.

File: spec/acceptance/06_postgresql_spec.rb (hunk 2 above)
Comment (treydock) [reply spec/acceptance/06_postgresql_spec.rb]: I added basic check that exits 0 to generate an event and check the event with and without postgresql support enabled.

File: .fixtures.yml (hunk 1 above)
Comment (ghoneycutt) [reply .fixtures.yml]: tag is not there.. forgot to push?

Comment (treydock) [reply .fixtures.yml]: Fixed, not sure where it ended up but also squashed commits

File: .fixtures.yml [hunk 3]
```diff
@@ -10,6 +10,18 @@ fixtures:
...
+      repo: git://github.com/puppetlabs/puppetlabs-augeas_core
+      ref: 1.0.4
+      puppet_version: ">= 6.0.0"
+    # Need by postgresql
+    concat:
+      repo: git://github.com/puppetlabs/puppetlabs-concat.git
+      ref: v6.0.0
+    postgresql:
+      repo: git://github.com/puppetlabs/puppetlabs-postgresql.git
+      ref: v6.0.0
```
Comment (ghoneycutt) [inline .fixtures.yml]: 6.1.0 - be sure to check all the new additions that we are using the latest release

File: README.md [hunk 4]
```diff
@@ -219,7 +222,40 @@ class { 'sensu::backend':
...
+**NOTE**: This features require a valid Sensu Go enterprise license.
+
+The following example will add a PostgreSQL server and database to the sensu-backend host and configure Sensu Go to use PostgreSQL as the event datastore.
+
+```puppet
+class { 'postgresql::globals':
+  manage_package_repo => true,
+  version             => '9.6',
+}
+class { 'postgresql::server': }
```
Comment (ghoneycutt) [inline README.md]: Is there a reason to use `class{}` with no params instead of `include` ?

File: spec/acceptance/sensu_check_spec.rb [hunk 5]
```diff
@@ -68,13 +68,6 @@
         expect(data['metadata']['namespace']).to eq('test')
       end
     end
-
```
Comment (ghoneycutt) [inline spec/acceptance/sensu_check_spec.rb]: why would this change? it seems unrelated.

**Comment by ghoneycutt** [comment]
This should also have a manifest for vagrant so it can easily be consumed there.

File: .fixtures.yml (hunk 3 above)
Comment (treydock) [reply .fixtures.yml]: This file has historically been the lower bound of version range, we test against the upper bound in `.fixtures-latest.yml`.

File: README.md (hunk 4 above)
Comment (treydock) [reply README.md]: Because that's what puppetlabs documents. Also ensuring we don't do something to break the ability for people to use `class { } ` since that's easier to break than just `include`.

File: spec/acceptance/sensu_check_spec.rb (hunk 5 above)
Comment (treydock) [reply spec/acceptance/sensu_check_spec.rb]: Because it's fragile. We add/remove checks in other files and to constantly having to update this number based on unrelated tests seems like a bad idea. Also we are directly testing that the checks we created all exist so checking the count just adds fragility to the acceptance tests.

**Comment by treydock** [comment]
We can't use Vagrant because postgresql support requires a license, so for someone to use it in Vagrant would require they supply a license file.

**Comment by treydock** [comment]
Do you want me to make it so an environment variable can define the path to license file or something? We don't currently have any enterprise licensed support in the vagrant images.

File: spec/acceptance/sensu_check_spec.rb (hunk 5 above)
Comment (treydock) [reply spec/acceptance/sensu_check_spec.rb]: This became an issue when I added checks to test the postgresql still worked and so an external test to this file was impacting this test which sort of violates the idea that tests are self contained.
